google/cloud/ndb/_transaction.py

```diff
@@ -23,6 +23,77 @@
 log = logging.getLogger(__name__)
 
 
+class Propagation(object):
+    def __init__(self, propagation, join, *args, **kwargs):
+        # Avoid circular import in Python 2.7
+        from google.cloud.ndb import context as context_module
+
+        propagation_options = context_module.TransactionOptions._PROPAGATION
+        if propagation is None or propagation in propagation_options:
+            self.propagation = propagation
+        else:
+            raise ValueError('Unexpected value for propagation. Got: ' \
+                             f'{propagation}. Expected one of: ' \
+                             f'{propagation_options}')
+
+        propagation_names = context_module.TransactionOptions._INT_TO_NAME
+        self.propagation_name = propagation_names.get(self.propagation)
+
+        self.join = join
+        joinable_options = context_module.TransactionOptions._JOINABLE
+        self.joinable = propagation in joinable_options
+        super(Propagation, self).__init__(*args, **kwargs)
+
+    def _handle_nested(self):
+        raise exceptions.NoLongerImplementedError()
+
+    def _handle_mandatory(self):
+        if not in_transaction():
+            raise exceptions.TransactionFailedError('Unable to honor ' \
+                'requested propagation option, an existing transaction must ' \
```

Comment:
We prefer less wordy message errors, if possible. Maybe something like "this option requires an existing transaction".